### PR TITLE
new graph-log-sketch scaffolding

### PR DESCRIPTION
**Pull request by meyerzinn**
@AdityaAtulTewari requested no trunk branch, so this obliterates the main branch and adds the scaffolding for our new edit scalability benchmark.

@anubhavgoel26 note that this changes the CLI flag options used for `algo` and `graph`.

### CODE REVIEW

**Comment by meyerzinn** [comment]
@AdityaAtulTewari I would like to land this before any more work goes into the repo since changes could quickly get very confusing. Lmk when you are comfortable merging.

File: include/graph.hpp
Comment (AdityaAtulTewari) [inline include/graph.hpp:1]: This file needs a more descriptive name.

File: include/algo.hpp
Comment (AdityaAtulTewari) [inline include/algo.hpp:1]: In a separate file please. Folder `include/algos/bfs.hpp` and `include/algos/nop.hpp`

File: microbench/edit_scalability.cpp [hunk 1]
```diff
@@ -1,121 +1,193 @@
...
+               cl::values(clEnumValN(lscsr, "lscsr", "log-structured CSR"),
+                          clEnumValN(morph, "morph", "Galois MorphGraph")));
+
+/*
+ * Algo args
+ */
+
+enum AlgoName { nop, sssp_bfs };
+
+cl::opt<AlgoName> algo_name(
```
Comment (AdityaAtulTewari) [inline microbench/edit_scalability.cpp:55]: Let's use a different library for our command line arguments, using llvm has caused a lot of problems. Use getopt.

File: include/graph.hpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+/* Contains the minimal functionalities that are benchmarked. */
+class Graph {
+public:
+  virtual int add_edges(uint64_t src, const std::vector<uint64_t> dsts) = 0;
+  virtual uint64_t size() noexcept                                      = 0;
+
+  virtual void for_each_edge(uint64_t src,
+                             std::function<void(uint64_t const&)> callback) = 0;
+};
```
Comment (AdityaAtulTewari) [inline include/graph.hpp:19]: This needs to be titled `EditableGraph` or smthing like that. The `Graph` has the api as described in other places. I would also prefer use of `CRTP`.

**Review by AdityaAtulTewari — CHANGES_REQUESTED** [review]

File: microbench/edit_scalability.cpp (hunk 1 above)
Comment (meyerzinn) [reply microbench/edit_scalability.cpp:55]: no pls

File: include/graph.hpp (hunk 2 above)
Comment (meyerzinn) [reply include/graph.hpp:19]: discussed this, putting CRTP off for now since it complicates the benchmarking code for x86

**Review by AdityaAtulTewari — APPROVED** [review]
lgtm

**Pull request merged**